Sources/Build/BuildOperation.swift

```diff
@@ -734,69 +734,38 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
     private func plan(subset: BuildSubset? = nil) throws -> (description: BuildDescription, manifest: LLBuildManifest) {
         // Load the package graph.
         let graph = try getPackageGraph()
-        let buildToolPluginInvocationResults: [ResolvedModule.ID: (target: ResolvedModule, results: [BuildToolPluginInvocationResult])]
-        let prebuildCommandResults: [ResolvedModule.ID: [PrebuildCommandResult]]
-        // Invoke any build tool plugins in the graph to generate prebuild commands and build commands.
-        if let pluginConfiguration, !self.config.shouldSkipBuilding(for: .target) {
+
+        let pluginTools: [ResolvedModule.ID: [String: PluginTool]]
+        // FIXME: This is unfortunately but we need to build plugin tools upfront at the moment.
```

Comment:
I think it would be helpful to future readers to explain why

Reply:
Sure, I will add a comment that explains it.

Reply:
@rauhul how does this look?